common/types/int.go

```diff
@@ -75,14 +86,32 @@ func (i Int) Compare(other ref.Val) ref.Val {
 func (i Int) ConvertToNative(typeDesc reflect.Type) (interface{}, error) {
 	switch typeDesc.Kind() {
 	case reflect.Int32:
+		// Enums are also mapped as int32 derivations.
 		return reflect.ValueOf(i).Convert(typeDesc).Interface(), nil
 	case reflect.Int64:
 		return int64(i), nil
 	case reflect.Ptr:
-		if typeDesc == jsonValueType {
+		switch typeDesc {
+		case anyValueType:
+			return ptypes.MarshalAny(&wrapperspb.Int64Value{Value: int64(i)})
+		case int32WrapperType:
+			return &wrapperspb.Int32Value{Value: int32(i)}, nil
+		case int64WrapperType:
+			return &wrapperspb.Int64Value{Value: int64(i)}, nil
+		case jsonValueType:
+			if i.isInt32() {
```

Comment:
This violates the convention used by proto3 where the number/decimal string choice is made based on the type, not the value.  You can still get JSON numbers if you're working with a message with the 32-bit integer field.

Reply:
I'm not sure I follow. We don't have an int32 internally, so this test is to comply with the spirit of documented conversion of int32 as float64 and int64 as string

Reply:
The documented conversion changes all integers from a 64-bit type to decimal strings, even those that fit in 53 or 32 bits, e.g. "1".  Our int type is 64-bit.  Yes, we do in some cases widen a 32-bit source to 64 bits, but at this point in the code we don't have that context.  None of the proto-to-JSON integer conversions depend on the values, so this is an innovation.

That being said, it's quite possible that most code which deals with the JSON-converted data is resigned to handling strings or numbers for fields which expect an int.  But if that's the case and we're going to make the JSON type depend on the value, then we might as well convert every int up to 2^53 (or 52? and there's a +/- 1 in there too) to a number.

Reply:
Ah, fair point. I think it makes sense to adjust to `int53` as the limit.